### ISSUE TITLE
support \MakeUppercase

### ISSUE THREAD

**Issue by xionghuichen** [ppizarror/PyDetex#33]
**Is your feature request related to a problem? Please describe.**

"I am \MakeUppercase{Happy}" is translated to  "I am".

**Describe the solution you'd like**
The problem is similar to  https://github.com/ppizarror/PyDetex/issues/28. Maybe we should design a default strategy to decide what to keep and what to remove.

**Describe alternatives you've considered**

add \MakeUppercase to the whitelist is the simplest solution.

**Comment by ppizarror** [comment]
The example ``"I am \MakeUppercase{Happy}" => "I am"`` should be translated to ``"I am \MakeUppercase{Happy}" => "I am HAPPY"``. Thus, we should add the following:

```latex
\uppercase{text}
\lowercase{text}
\MakeUppercase{text}
\MakeLowercase{text}
```

To add these methods, the function https://github.com/ppizarror/PyDetex/blob/3b39b084ec715751631965508674e918ea07ea28/pydetex/parsers.py#L702-L787 must be edited. I was thinking of allowing functions/lambdas in ``commands``:

```python
commands = [...,
('MakeUppercase', [(1, False)], lambda s: s.upper(), 1, 'normal', 'normal', (False, False))
]
```

In that case, line https://github.com/ppizarror/PyDetex/blob/3b39b084ec715751631965508674e918ea07ea28/pydetex/parsers.py#L778-L779 should support methods instead of calling ``string.format``, for example:

```python
if callable(cmd_tag):
    text = cmd_tag(*args)
else:
    text = cmd_tag.format(*args)
```

With such methods, I think we could easily add more methods in the future. This solution differs from #28 because of the nature of the input. Citeauthor requires cites and a different font type; in this case, it only takes one arg of "normal" text.

Let me know if you can add these methods =). Greetings

**Comment by xionghuichen** [comment]
Thanks for the response. I have not tried the solution yet, but it looks reasonable. I will reply to you as soon as possible after I tried it. :)

**Comment by ppizarror** [comment]
Implemented in #35.
